espaloma/graphs/deploy.py

```diff
@@ -0,0 +1,171 @@
+
+# =============================================================================
+# MODULE FUNCTIONS
+# =============================================================================
+def openmm_system_from_graph(g, forcefield="openff_unconstrained-1.1.0", suffix=""):
+    """ Construct an openmm system from `espaloma.Graph`.
+
+    Parameters
+    ----------
+    g : `espaloma.Graph`
+        Input graph.
+
+    Returns
+    -------
+    sys : `openmm.System`
+        Constructed single-molecule OpenMM system.
+
+
+    """
+    # get a forcefield
+    try:
+        ff = ForceField("%s.offxml" % forcefield)
+    except:
+        try:
+            ff = ForceField("test_forcefields/%s.offxml" % forcefield)
+        except:
+            raise NotImplementedError
+
+    # get the mapping between position and indices
+    bond_lookup = {
+        tuple(idxs.detach().numpy()): position
+        for position, idxs in enumerate(g.nodes['n2'].data['idxs'])
+    }
+
+    angle_lookup = {
+        tuple(idxs.detach().numpy()): position
+        for position, idxs in enumerate(g.nodes['n3'].data['idxs'])
+    }
+
+    # torsion_lookup = {
+    #     tuple(idxs.detach().numpy()): position
+    #     for position, idxs in enumerate(g.nodes['n4'].data['idxs'])
+    # }
+
+    sys = ff.create_openmm_system(g.mol.to_topology())
+
+    for force in sys.getForces():
+        name = force.__class__.__name__
+        if 'HarmonicBondForce' in name:
+            for idx in range(force.getNumBonds()):
+                idx0, idx1, eq, k = force.getBondParameters(idx)
+                position = bond_lookup[(idx0, idx1)]
+                _eq = g.nodes['n2'].data['eq%s' % suffix][position].detach(
+                    ).numpy().item()
+                _k = g.nodes['n2'].data['k%s' % suffix][position].detach(
+                    ).numpy().item()
+
+                _eq = Quantity(
+                    _eq,
+                    esp.units.DISTANCE_UNIT,
+                ).value_in_unit(OPENMM_BOND_EQ_UNIT)
+
+                _k = 2.0 * Quantity(
+                    _k,
+                    esp.units.FORCE_CONSTANT_UNIT,
+                ).value_in_unit(OPENMM_BOND_K_UNIT)
+
+                force.setBondParameters(idx, idx0, idx1, _eq, _k)
+
+        if 'HarmonicAngleForce' in name:
+            for idx in range(force.getNumAngles()):
+                idx0, idx1, idx2, eq, k = force.getAngleParameters(idx)
+                position = angle_lookup[(idx0, idx1, idx2)]
+                _eq = g.nodes['n3'].data['eq%s' % suffix][position].detach(
+                    ).numpy().item()
+                _k = g.nodes['n3'].data['k%s' % suffix][position].detach(
+                    ).numpy().item()
+
+                _eq = Quantity(
+                    _eq,
+                    esp.units.ANGLE_UNIT,
+                ).value_in_unit(OPENMM_ANGLE_EQ_UNIT)
+
+                _k = 2.0 * Quantity(
+                    _k,
+                    esp.units.ANGLE_FORCE_CONSTANCE_UNIT,
```

Comment:
`constance` --> `constant`

Reply:
Ops

Reply:
Looks like I made a typo in the original script and atom auto-completed for me.